crates/wasmparser/src/validator/operators.rs

```diff
@@ -252,62 +252,89 @@ impl OperatorValidator {
     /// simply that something is needed to be popped.
     ///
     /// If `expected` is `Some(T)` then this will be guaranteed to return
-    /// `Some(T)`, and it will only return success if the current block is
+    /// `T`, and it will only return success if the current block is
     /// unreachable or if `T` was found at the top of the operand stack.
     ///
     /// If `expected` is `None` then it indicates that something must be on the
     /// operand stack, but it doesn't matter what's on the operand stack. This
     /// is useful for polymorphic instructions like `select`.
     ///
     /// If `Some(T)` is returned then `T` was popped from the operand stack and
-    /// matches `expected`. If `None` is returned then it means that `None` was
+    /// matches `expected`. If `Bot` is returned then it means that `None` was
```

Comment:
If `T` is returned

Also: should we pass expected: Bot instead of having it be optional?